src/waveform/waveformwidgetfactory.cpp

```diff
@@ -432,6 +441,22 @@ void WaveformWidgetFactory::setZoomSync(bool sync) {
     }
 }
 
+void WaveformWidgetFactory::setDisplayBeatGrid(bool sync) {
+    m_beatGridEnabled = sync;
+    if (m_config) {
+        m_config->set(ConfigKey("[Waveform]", "beatGridLinesCheckBox"), ConfigValue(m_beatGridEnabled));
```

Comment:
I think it would make more sense to set the ConfigKey in DlgPrefWaveform than here.